core/src/main/java/com/graphhopper/routing/subnetwork/EdgeBasedTarjanSCC.java

```diff
@@ -46,7 +45,7 @@
  */
 public class EdgeBasedTarjanSCC {
     private final Graph graph;
-    private final Weighting weighting;
+    private final EdgeTransitionFilter edgeTransitionFilter;
```

Comment:
I wanted to reduce the dependency between the tarjan algorithm and weighting. Now I have this interface which specifies whether or not an edge-transition is allowed or not, similarly to node-based tarjan where we specify whether an edge is allowed or not (using EdgeFilter)

Reply:
Good idea. What about deprecating EdgeFilter and use this new more powerful interface instead in general?

Reply:
But maybe for a separate PR :)

Reply:
Hm the 'transition' part is not always needed. For example `LocationIndex#findClosest(lat, lon, EdgeFilter)` or `Graph#createEdgeExplorer(EdgeFilter)` are just filtering edges without any context what the 'previous' edge was.